dbms/src/DataStreams/TiRemoteBlockInputStream.h

```diff
@@ -155,21 +159,25 @@ class TiRemoteBlockInputStream : public IProfilingBlockInputStream
             }
 
             const auto & decode_detail = result.decode_detail;
+            total_rows += decode_detail.rows;
+
+            if (!result.rowsInfoOnly)
+            {
+                size_t index = 0;
+                if constexpr (is_streaming_reader)
+                    index = result.call_index;
 
-            size_t index = 0;
-            if constexpr (is_streaming_reader)
-                index = result.call_index;
+                ++connection_profile_infos[index].packets;
```

Comment:
```suggestion
                connection_profile_infos[index].packets += decode_detail.packets;
```
And then we don't need to use rowsInfoOnly.

Reply:
Finally, sacrified the accurate of index info, use newOk instead of newRowsInfoOnly to keep protocal simple.